stolinski/svelte-5-drizzle-sveltekit-2-example
# Update Habit

## The Issue

As a user, I want to be able to edit a habit

## The Solution

@stolinski I picked this up since it was on the roadmap. If this implementation is not what you envisioned, please feel free to provide feedback or reject the PR 😀

- Add an "edit" action for each habit row
- Refactor the `NewHabitForm` to allow for creating or editing
  - Perhaps this is not the best approach. An alternative solution I considered was to create a second form called `EditHabitForm`. I'm happy to make that revision if necessary.
- Refactor the server actions for creating/editing habits so the validation code can be shared between both
- Fix the `Reorder` svg so the proper color is filled (it had a semicolon after the fill value)

## To Test

- Create a habit
- Edit the habit
- Make sure validation still works as expected

## Visuals

### In Motion

![after](https://github.com/stolinski/svelte-5-drizzle-sveltekit-2-example/assets/20483201/404c44bf-eda7-4111-9ace-a1d0a4314e74)

### Reorder Icon Fix
Before | After
----- | -----
![image](https://github.com/stolinski/svelte-5-drizzle-sveltekit-2-example/assets/20483201/9a87ef27-b1bf-4d23-9fa0-2478af130fbd) | ![image](https://github.com/stolinski/svelte-5-drizzle-sveltekit-2-example/assets/20483201/02f789f8-a9c2-4125-a22b-7560d840df3e)
